### DISCUSSION TITLE
How to modify the color of the image in show3d?

### DISCUSSION THREAD

**Discussion by Xiaozizhuo** [PMEAL/porespy#691] [Q&A]
I want to change the color of the model to make the model more suitable for my ppt, but I don't know where the relevant code is.
（Examples come from https://porespy.org/examples/generators/tutorials/cylinders.html   I only modified the dimension parameters of the model）
![GDL半径为2孔隙度为0 8](https://user-images.githubusercontent.com/98071787/166145600-9393fd22-acb6-45cf-bcd7-a6607c0e55ef.png)

And how do you generate that code into a vtk file and import it into paraview?

**Comment by jgostick** [comment]
Hi @Xiaozizhuo 
The show3D function is meant for quick and dirty checks of the struture because opening in paraview is a bit of a pain.  However, as of about 2 days ago, I became aware of a new package called vedo.  I have put a quick code snippet at the end of [this thread](https://github.com/PMEAL/porespy/discussions/557) (#557).  You just need to ``pip install vedo`` and it just works!  

If you still want to do paraview, which is for sure the most powerful, then the solution is easy, just do ``imageio.volsave('file.tif', np.array(im, dtype=int))``, then open the tif file in paraview directly.

**Reply by jgostick** [reply]
I should also add this link to an example: https://porespy.org/examples/visualization/tutorials/visualizing_tif_in_paraveiw.html

This example does not include how to 'threshold' the image to only show the solids, but there is a threshold icon on the paraview menu bar, so move the sliders to control which values are shown or not.

**Reply by Xiaozizhuo** [reply]
Sorry professor, why not use the functions of the io module in poultespy to create a vtk file?  However, I wanted to implement this method, but encountered some problems.
For example, 
![VTK](https://user-images.githubusercontent.com/98071787/166177592-63f538a7-0f1f-4566-a449-2d7cf3e309ec.png)

But I don't know, in my code, what im should fill in
![vtk](https://user-images.githubusercontent.com/98071787/166177736-dc7a1a5d-44ee-4ad0-af69-a1a28fc23bad.png)

**Reply by jgostick** [reply]
I must confess that I have not used this function myself, I always just use the the 'tif' image trick.  The docstring looks pretty clear though.  What problem are you having exactly?

**Reply by Xiaozizhuo** [reply]
The situation is like this, in the to_vtk function you need to fill in im and filename, but I don't know who im should be filled in in the above code.

**Reply by Xiaozizhuo** [reply]
Hey Professor, I studied your code carefully to understand the meaning of the im parameter, and I have successfully generated the vtk file and can be opened in paraview, but only opened, and after opening it is just an empty cube box.

**Reply by jgostick** [reply]
That's good to hear.  I think the issue of it being an empty box is just learning how to use paraview.  You probably have to add glyphs or change the display somehow.
